src/Raven.Server/Web/Studio/StudioCollectionsHandlerProcessorForPreviewRevisions.cs

```diff
@@ -18,18 +20,38 @@ internal sealed class StudioCollectionsHandlerProcessorForPreviewRevisions : Abs
 
     public StudioCollectionsHandlerProcessorForPreviewRevisions([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
     {
+        _start = RequestHandler.GetStart();
+        _pageSize = RequestHandler.GetPageSize();
     }
 
-    protected override async Task InitializeAsync(DocumentsOperationContext context, CancellationToken token)
+    protected override Task InitializeAsync(DocumentsOperationContext context, CancellationToken token)
     {
-        await base.InitializeAsync(context, token);
-
-        _start = RequestHandler.GetStart();
-        _pageSize = RequestHandler.GetPageSize();
+        switch (Type)
+        {
+            case RevisionsStorage.RevisionsType.All:
+                _totalResults = string.IsNullOrEmpty(Collection)
+                    ? RequestHandler.Database.DocumentsStorage.RevisionsStorage.GetNumberOfRevisionDocuments(context)
+                    : RequestHandler.Database.DocumentsStorage.RevisionsStorage.GetNumberOfRevisionDocumentsForCollection(context, Collection);
+                break;
+
+            case RevisionsStorage.RevisionsType.NotDeleted:
+                _totalResults = string.IsNullOrEmpty(Collection)
+                    ? RequestHandler.Database.DocumentsStorage.RevisionsStorage.GetNumberOfNonDeletedRevisions(context)
+                    : -1; // Not available for specific collection
```

Comment:
make this a const and use it in all places

Reply:
Done